src/Symfony/Component/Process/Process.php

```diff
@@ -443,6 +452,8 @@ public function disableOutput()
     public function enableOutput()
     {
         if ($this->isRunning()) {
+            $this->stop(0);
```

Comment:
same here and below.

In an exception-driven workflow, you could:
```
$process = new Process();
$process->disableOutput();

// here this service try to access output like described in the following block
$service->handle($process);
```
```
// service
   function handle(Process $process) {
      try {
          $output = $process->getOutput();
      } catch (\RuntimeException) {
          $output = null;
      }
   }
```

This is a BC break because the promise has never been that when an exception is thrown, process is stopped. The promise is : when something is not possible let's throw an exception.